transform/new_resource_test.go

```
@@ -0,0 +1,271 @@
+	if patch == nil || len(patch) != 1 {
+		t.Errorf("expected 1 patch op (rules), got %v", patch)
+	}
+
+	// Verify roundtrip
+	skelJSON, _ := skeleton.MarshalJSON()
+	patched, err := patch.Apply(skelJSON)
+	if err != nil {
+		t.Fatalf("failed to apply patch: %v", err)
+	}
+	var result map[string]interface{}
+	json.Unmarshal(patched, &result)
```

golangci-lint (2.12.2)

[error] Error return value of `json.Unmarshal` is not checked

(errcheck)